http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -301,31 +306,70 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
-    for {
-      ctx <- ev.ask
-      diskOpt <- persistentDiskQuery.getById(params.runtimeConfig.persistentDiskId).transaction
-      disk <- F.fromOption(diskOpt, new RuntimeException("no disk found"))
-      common = getCommonFields(ControlledResourceName(disk.name.value),
-                               config.runtimeDefaults.diskControlledResourceDesc,
-                               params.runtime.auditInfo.creator,
-                               None
-      )
-      request: CreateDiskRequest = CreateDiskRequest(
-        params.workspaceId,
-        common,
-        CreateDiskRequestData(
-          // TODO: AzureDiskName should go away once DiskName is no longer coupled to google2 disk service
-          AzureDiskName(disk.name.value),
-          disk.size,
-          params.runtimeConfig.region
-        )
-      )
-      diskResp <- wsmDao.createDisk(request, leoAuth)
-      _ <- controlledResourceQuery
-        .save(params.runtime.id, diskResp.resourceId, WsmResourceType.AzureDisk)
-        .transaction
-      _ <- persistentDiskQuery.updateStatus(disk.id, DiskStatus.Ready, ctx.now).transaction
-    } yield diskResp
+    params.useExistingDisk match {
+      case true =>
+        for {
+          ctx <- ev.ask
+          diskOpt <- persistentDiskQuery.getById(params.runtimeConfig.persistentDiskId).transaction
+          disk <- F.fromOption(
+            diskOpt,
+            new RuntimeException(s"Disk id:${params.runtimeConfig.persistentDiskId} not found")
+          )
+          resourceId <- F.fromOption(
+            disk.wsmResourceId,
+            AzureRuntimeCreationError(
+              params.runtime.id,
+              params.workspaceId,
+              s"No associated resourceId found for Disk id:${params.runtimeConfig.persistentDiskId.value}",
+              params.useExistingDisk
+            )
+          )
+          diskResourceOpt <- controlledResourceQuery
+            .getWsmRecordFromResourceId(resourceId, WsmResourceType.AzureDisk)
+            .transaction
+          wsmDisk <- F.fromOption(
+            diskResourceOpt,
+            AzureRuntimeCreationError(
+              params.runtime.id,
+              params.workspaceId,
+              s"WSMResource:${resourceId} not found for disk id:${params.runtimeConfig.persistentDiskId.value}",
+              params.useExistingDisk
+            )
+          )
+          _ <- persistentDiskQuery.updateStatus(disk.id, DiskStatus.Ready, ctx.now).transaction
+          _ <- controlledResourceQuery
+            .updateRuntime(wsmDisk.resourceId, WsmResourceType.AzureDisk, params.runtime.id)
+            .transaction // set runtime to new runtimeId
+          diskResp = CreateDiskResponse(wsmDisk.resourceId)
+        } yield diskResp
+      case _ =>
+        for {
+          ctx <- ev.ask
+          diskOpt <- persistentDiskQuery.getById(params.runtimeConfig.persistentDiskId).transaction
+          disk <- F.fromOption(diskOpt, new RuntimeException("no disk found"))
```

Comment:
We typically avoid generic exceptions, and use something than inherits from `PubsubHandleMessageError`. For example `AzureRuntimeCreationError`. This allows us to use consistent error handling. I'd double check `messageHandler` in `LeoPubsubMessageSubscriber`, and just ensure the error is being passed to a handler you're happy with